dd-sdk-android/src/test/kotlin/com/datadog/android/rum/tracking/NavigationViewTrackingStrategyTest.kt

```diff
@@ -154,6 +154,17 @@ internal class NavigationViewTrackingStrategyTest {
         verify(mockNavController).addOnDestinationChangedListener(testedStrategy)
     }
 
+    @Test
+    fun `registers navigation listener when nav controller added after activity started`() {
+        // TODO reset mockActivity
```

Comment:
Those were to be completed, I'm sorry I might have not been clear enough. As is I'm not sure the unit test passes. 

Reset would be something like: 
```
whenever(mockNavView.getTag(R.id.nav_controller_view_tag)) doReturn null
```

and setup below would be 
```
whenever(mockNavView.getTag(R.id.nav_controller_view_tag)) doReturn mockNavController
```

Also make sure that all other tests in this class pass as well.

Reply:
Ok, I'll see what I can do - as said, unit tests refused to work on my machine (both `instrumented.integration` and `instrumented.nightly-tests`)

Reply:
Those are just unit tests, you should be able to run them with `gradlew uTD` or `gradlew :dd-sdk-android:testDebugUnitTest`

Reply:
Allright, updated unit tests as well (sorry, took me a while, can't take my full time on this).

With new code `unregisters navigation listener onActivityStopped` started failing because now `startedActivity` needs to be assigned, so:

- added unit test for dynamic setup (as discussed above)
- split `unregisters navigation listener onActivityStopped` into 2 unit tests:
- `unregisters navigation listener onActivityStopped if activity was started` verifies the same thing as original "unregister" test, except that it calls `onActivityStarted` as well
- `doesn't unregister navigation listener onActivityStopped if activity not started` takes care of use case where only `onActivityStopped` is called and hence no unregistration occurs

Technically, in real world usage, last case should not happen - `onActivityStopped` is only called after prior `onActivityStarted`. But the code is ready for this so I thought unit test should reflect it too.

Note: there are couple flaky unit tests (3-5) that fail on any test run for me. But they're the same as before my changes.

Let me know your further thoughts...